src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -433,6 +437,7 @@ private void PreInitialize()
 
                 if (string.IsNullOrEmpty(extensionVersion))
                 {
+                    _hostMetrics.AppFailure();
```

Comment:
Seems like we should handle failures centrally in the host init exception handler. Looks like you're doing that below so this line isn't needed?

Reply:
WebJobsScriptHostService.UnsynchronizedStartHostAsync is the main host loop responsible for handling startup failures and retires. See the logic there - seems like you should be doing your metering there.

Reply:
Just to clarify, the suggestion is to remove all records of "AppFailure();" currently sprinkled around the host, and to just have one occurance of this in UnsynchronizedStartHostAsync

Reply:
Is the goal of this new metric to capture any time a host level startup error occurs? I think those can happen in lots of places. However, I think the catch handler in UnsynchronizedStartHostAsync will end up catching all your cases so I'm not sure if you need to instrument all over (might lead to duplicate counting even). It's like a big try/catch around all these startup operations.

Reply:
Reduced all use of AppFailure() to just UnsynchronizedStartHostAsync.

> Is the goal of this new metric to capture any time a host level startup error occurs?

I think I answer this question here: https://github.com/Azure/azure-functions-host/pull/9730#discussion_r1471959622